### PR TITLE
Convert some components to ts && add image module

### PR DESCRIPTION
**Pull request by Snizhana**

### CODE REVIEW

File: tests/helpers/index.test.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@ describe('misc', () => {
     const snippet = {
       content: 'const memoizedSyntaxes = useMemo()',
       created_at: '2019-09-29T19:23:31',
-      id: 364066,
+      id: '364066',
```
Comment (Snizhana) [inline tests/helpers/index.test.ts:9]: [snippet type](https://github.com/xsnippet/xsnippet-web/blob/master/src/store/index.ts#L7) describes this prop as a string

File: tests/helpers/index.test.ts [hunk 2]
```diff
@@ -26,7 +26,7 @@ describe('misc', () => {
     })
 
     it('should return propper title', () => {
-      const untitled = { ...snippet, title: undefined }
+      const untitled = { ...snippet, title: '' }
```
Comment (Snizhana) [inline tests/helpers/index.test.ts:29]: [snippet type](https://github.com/xsnippet/xsnippet-web/blob/master/src/store/index.ts#L8) describes this prop as a string (required)

File: package.json [hunk 3]
```diff
@@ -51,13 +54,15 @@
...
+    "@types/enzyme": "^3.10.8",
+    "@types/jest": "^26.0.20",
     "@types/parse-link-header": "^1.0.0",
     "@types/react": "^17.0.0",
     "@types/react-dom": "^17.0.0",
+    "@types/react-router-dom": "^5.1.7",
     "@typescript-eslint/eslint-plugin": "^4.14.0",
     "@typescript-eslint/parser": "^4.14.0",
     "babel-core": "^7.0.0-bridge.0",
-    "babel-jest": "^23.4.2",
```
Comment (Snizhana) [inline package.json:60]: we use `ts-jest` instead

File: src/assets.d.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+declare module '*.jpg' {
```
Comment (Snizhana) [inline src/assets.d.ts:1]: in order to import images in ts, we need to declare related modules

**Comment by lotrien** [comment]
hey @Snizhana I think this pr is ready to go

**Review by lotrien — APPROVED** [review]

**Pull request merged**